enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,973 @@
+      * Add `node` as the first command line arg.
+      * This allows us not to give the DaemonSet any AWS credentials.
+  * Only on HyperShift:
+    * When interacting with Kubernetes API, carefully choose the right kubeconfig to watch / create / update objects in
+      the right cluster.
+    * Replace namespaces in the driver Deployment, PDBs and other objects that are created in the management cluster.
+      They must be created in the same namespace as the operator.
+    * Update the CSI driver control-plane Deployment: 
+      * Give it kubeconfig to the guest cluster as a Secret volume.
+      * Pass it Secret with STS creds generated by control-plane-operator.
+      * Add `token-minter` sidecar, which will create STS secrets to the driver.
+      * Set `priorityClassName: hypershift-control-plane`.
+    * Phase 2: add metrics services + TLS keys.
+
+* **AWS EBS CSI driver**:
+  * No changes there!
+    * Controller parts will run in the management cluster, but they get kubeconfig to the guest cluster.
+    * Node DaemonSet will get the default projected kubeconfig that can access objects only in the
+      openshift-cluster-csi-drivers namespace. 
+
+#### Notes about other CSI drivers
+
+* Azure Disk CSI driver
+  * It needs some cloud credentials available on nodes. We don't have exact list of permissions needed, they could be
+    very minimal, still, it needs to be there.
+* vSphere CSI driver
+  * Controller parts of the CSI driver assume that they run in vSphere. Significant driver changes may be needed
+    to be able to run the controller parts on any cloud or on bare metal.
+  * vSphere is often behind a firewall and is often not accessible from outside of vSphere itself. I.e. if the
+    management cluster is outside of vSphere, it may be necessary to open new firewall ports.  
+* On vSphere, cluster-storage-operator installs vsphere-problem-detector. It's not clear if it should run in the
+  management cluster or in the guest cluster or in both. Because both cluster will need to talk to vSphere API
+  and the problem detector is there to check for common problems in the connection.
+
+### Risks and Mitigations
+
+#### Storage CRs are in the guest cluster
+
+* Since CRs for cluster-storage-operator, aws-ebs-csi-driver-operator and cluster-csi-snapshot-controller-operator are
+  in the guest cluster, a guest cluster-admin user can DOS the whole CSI driver / CSI snapshots by editing these CRs in
+  an endless loop. The operators will redeploy the operands with each CR change and the new operands will need to wait
+  for leader election to expire, while the old operands are already terminated.
+  * We mitigate this issue a bit by overwriting user changes in the operator CRs by Hosted Cluster Config Operator that
+    runs in HyperShift management cluster. Still, an evil cluster-admin user can easily write a loop that updates the
+    CRs in a busy loop.
+
```

Comment:
how would lease in management would mitigate this? is there a plan for it?

Reply:
Lease in management cluster won't do anything useful here.
The next bullet talks about leases, but then we would need to change upstream CSI components to handle two kubeconfigs, one for leases (in the mgmt cluster) and second for the real storage objects. I addded a note about this.